geniza/corpus/models.py

```diff
@@ -26,15 +29,21 @@ class Collection(models.Model):
     class Meta:
         ordering = ['abbrev']
         constraints = [
-            models.UniqueConstraint(fields=['library', 'collection'],
-                                    name='unique_library_collection')
+            # require at least one of library OR name
+            models.CheckConstraint(
+                check=(models.Q(library__regex='.+') |
+                       models.Q(name__regex='.+')),
+                name='req_library_or_name'
+            ),
```

Comment:
weird that this requires a regex, but glad we now have an example of how to do it!

Reply:
I couldn't figure out any other way to look for non-empty content! Agree it is kind of weird.